edx_api/enrollments/init_test.py

```diff
@@ -31,23 +31,93 @@ def setUpClass(cls):
                                'fixtures/enrollments_list.json')) as file_obj:
             cls.enrollments_list_json = json.loads(file_obj.read())
 
-        cls.enrollment_json = cls.enrollments_json[0]
         base_edx_url = 'http://edx.example.com'
         cls.base_url = urljoin(base_edx_url, CourseEnrollments.enrollment_url)
-        client = EdxApi({'access_token': 'foobar'}, cls.base_url)
-        cls.enrollment_client = client.enrollments
+        cls.client = EdxApi({'access_token': 'foobar'}, cls.base_url)
+        cls.enrollment_client = cls.client.enrollments
 
     @requests_mock.mock()
     def test_create_enrollment(self, mock_req):
         """
         Tests the post request to create an enrollment.
         This just tests that the client expects a JSON object representing the enrollment.
         """
-        mock_req.post(self.base_url, text=json.dumps(self.enrollment_json))
-        enrollment = self.enrollment_client.create_audit_student_enrollment(
-            self.enrollment_json['course_details']['course_id'])
-        for key, val in enrollment.json.items():
-            assert self.enrollment_json.get(key) == val
+        mock_req.register_uri(
+            'POST',
+            self.base_url,
+            [{'json': self.enrollments_json[0], 'status_code': 200},
+             {'json': self.enrollments_json[1], 'status_code': 200}]
+        )
+
+        course_id = self.enrollments_json[0]['course_details']['course_id']
+        enrollments = []
+
+        audit_enrollment = self.enrollment_client.create_student_enrollment(
+            course_id,
+            mode="honors",
+            user="user"
+        )
+        enrollments.append(audit_enrollment)
+
+        honors_enrollment = self.enrollment_client.create_audit_student_enrollment(course_id)
+        enrollments.append(honors_enrollment)
+
+        for enrollment, enrollment_dict in zip(enrollments, self.enrollments_json):
+            self.assertDictEqual(enrollment.json, enrollment_dict)
```

Comment:
This test case is doing a bunch of things it doesn't need to do. First I would define an extra property in `setUpClass` for this mock response:

```python
    @classmethod
    def setUpClass(cls):
        with open(os.path.join(os.path.dirname(__file__),
                               'fixtures/user_enrollments.json')) as file_obj:
            cls.enrollments_json = json.loads(file_obj.read())
        cls.enrollment_responses = [
            {'json': enrollment_json, 'status_code': 200}
            for enrollment_json in cls.enrollments_json
        ]
        #...
```

After that, this test case can be reduced to this:

```python
        mock_req.register_uri('POST', self.base_url, self.enrollment_responses)
        course_id = 'dummy_course_id'
        returned_enrollments = [
            self.enrollment_client.create_student_enrollment(course_id),
            self.enrollment_client.create_audit_student_enrollment(course_id)
        ]
        assert [
            enrollment.json for enrollment in returned_enrollments
        ] == self.enrollments_json
```

I would also change this test case name to `test_create_enrollment_return_val`